bcgov/HOUS-permit-portal
# Add ability to generate step code from h2k file

[BPHH-431]

## Description

Adds a step code tab to the nav (temporary - this will move into the submitter flow eventually). This allows the submitter to create a new step code from a hot2000 file and then add in additional fields. There is currently no validation on fields and there are some missing permit application fields under project info and plan info.

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[BPHH-431](https://hous-bssb.atlassian.net/browse/BPHH-431)
[BPHH-678)(https://hous-bssb.atlassian.net/browse/BPHH-678)
[BPHH-214](https://hous-bssb.atlassian.net/browse/BPHH-214)

## Steps to QA

pre reqs:
- install minio and run server (see README)

1. log in as a submitter
2. click on "Step Codes" in sub nav
3. create a new step code with a h2k file (see [BPHH-214](https://hous-bssb.atlassian.net/browse/BPHH-214) for test h2k files)
4. add any additional checklist fields and save

## UI Accessibility Checklist

Will do this on final UI after mocks are implemented

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

Pre Deploy: Add ENV variables as per .env.example
